### PR TITLE
jenkins: improve x64 node.exe caching for arm64

### PR DESCRIPTION
**Pull request by StefanStojanovic**
It was noticed on a few occasions that x64 node.exe caching is error-prone. This happens due to issues with downloading the new node.exe which, if it goes bad, makes the machine unusable for arm64 compilation for a week. This improvement in caching logic removes that possibility, by validating the downloaded file's SHASUM and replacing the old one with it, only if it matches the SHASUM found in the official `SHASUMS256.txt` file.

Fixes: https://github.com/nodejs/build/issues/3504
Refs: https://github.com/nodejs/build/pull/3315

### CODE REVIEW

**Comment by StefanStojanovic** [comment]
Can someone from @nodejs/build review please? After landing this, the same code will be added to the release CI since not having caching there caused machines to be stuck on ARM64 builds indefinitely until manual action is taken to unblock it.

**Review by richardlau — APPROVED** [review]

**Pull request merged**
